telescope/selector.py

```diff
@@ -87,25 +91,70 @@ def _parse_file_contents(self, selector_file_contents):
     selector_input_json = json.loads(selector_file_contents)
     self.validate_selector_input(selector_input_json)
 
-    metrics = []
     if selector_input_json['metric'] == 'all':
-      metrics.extend(self.supported_metrics.keys())
-    else:
-      metrics.append(selector_input_json['metric'])
+      selector_input_json['metric'] = self.supported_metrics.keys()
+
+    selector_dicts = self._parse_input_for_selectors(selector_input_json)
+    selectors = self._create_selectors_from_dict(selector_dicts)
+    
+    return selectors
 
+  def _parse_input_for_selectors(self, selector_dict):
+    """ Parse the selector JSON diction and return a list of dictionaries
+      flattened for each combination.
+      
+      Args:
+        selector_dict (dict): Dictionary parsed from a valid
+          selector JSON file with potentially lists for values.
+      
+      Returns:
+        list: List of dictaries representing the possible combinations of
+          the selector query.
+      
+    """
     selectors = []
+    has_not_recursed = True
+    
+    for iterable_field_key in self.iterable_field_keys:
```

Comment:
I think it's possible to do this with less complexity using `itertools`. Could we do something like this:

```
start_times = selector_json['start_time']
client_providers = selector_json['client_provider']
sites = selector_json['site']
metrics = selector_json['metric']

for start_time, client_provider, site, metric in itertools.product(start_times, client_providers, sites, metrics):
  selector = Selector()
  selector.start_time = start_time
  ...
```

That way we don't have to deal with making lots of copies of the JSON or passing around these sort of "not quite real `Selector` objects yet" selector `dicts`.